tests/unit_tests/data/test_geometry.py

```diff
@@ -46,15 +46,36 @@
                 (0.1, 1.1),
             ],
         ),
+        (
+            [(-10, 6.5), (-10, -6.5), (6, -6.5), (6, 6.5)],
+            [
+                [
+                    (0, 6.5),
+                    (0, 0.35),
+                    (0.2, 0.35),
+                    (0.2, 0.85),
+                    (2.2, 0.85),
+                    (2.2, 6.5),
+                ],
+                [
+                    (0, -6.5),
+                    (0, -0.35),
+                    (0.2, -0.35),
+                    (0.2, -0.85),
+                    (2.2, -0.85),
+                    (2.2, -6.5),
+                ],
+            ],
+        ),
     ],
 )
 def test_create_walkable_area_from_coordinates(point_input):
     reference_polygon = shapely.Polygon(*point_input)
+    if not reference_polygon.is_valid:
```

Comment:
This reference_polygon is not valid, right? 
Why do you check validity here? And why is it manipulated later with buffer 0?